libs/testdiff/golden.go

```diff
@@ -122,6 +123,15 @@ func (r *ReplacementsContext) Set(old, new string) {
 	if old == "" || new == "" {
 		return
 	}
+
+	encodedNew, err := json.Marshal(new)
```

Comment:
Please include a comment explaining why this is done.

Clear from the PR but not inline for a future reader.

Also, this is guaranteed to be a subset of the general case, right?

Reply:
Added a comment https://github.com/databricks/cli/pull/2142/commits/1fd0837bb4b501d604402e5b3ae97263060b02e3

> Also, this is guaranteed to be a subset of the general case, right?

Did not get this question.

Reply:
As in, the quoted copy is more specific than the non-quoted one, so it must be replaced before the non-quoted one to not interfere.

Reply:
Yes, that should be the case.